infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -185,6 +204,45 @@ public void startAndListen() {
         this.start().channel().closeFuture().syncUninterruptibly();
     }
 
+    public void restartServerChannel() {
+        log.info("restartServerChannel: Stopping Corfu Server channels.");
+
+        if (bindFuture != null) {
+            bindFuture.channel().close().syncUninterruptibly();
+        }
+
+        if (httpServerFuture != null) {
+            httpServerFuture.channel().close().syncUninterruptibly();
+        }
```

Comment:
Although it doesn't seem like a common possibility, there's a potential visibility issue here since these futures are set by a different thread than the one executing this method. Should these be made volatile?

Reply:
This could probably be useful for weak infra setups. Done, Thanks!